### ISSUE TITLE
XCode Build Failure

### ISSUE THREAD

**Issue by zeninfinity** [OpenSpace/OpenSpace#397]
I am attempting to build OpenSpace in Xcode (on OSX) and get errors while building.

When running ccmake I get the following errors:

```INFO RT-extension not found. glTF import/export will be built without Open3DGC-compression.

 CMake Warning (dev) at CMakeLists.txt:184 (set):
   Cannot set "CMAKE_PREFIX_PATH": current scope has no parent.
 This warning is for project developers.  Use -Wno-dev to suppress it.


 CMake Warning at /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:767
 (message):
   Imported targets not available for Boost version 106501
 Call Stack (most recent call first):
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:871
 (_Boost_COMPONENT_DEPENDENCIES)
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:1501
 (_Boost_MISSING_DEPENDENCIES)
   apps/Launcher/ext/libtorrent/CMakeLists.txt:190 (find_package)



 CMake Warning at /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:767
 (message):
   Imported targets not available for Boost version 106501
 Call Stack (most recent call first):
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:871
 (_Boost_COMPONENT_DEPENDENCIES)
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:1501
 (_Boost_MISSING_DEPENDENCIES)
   apps/Launcher/ext/libtorrent/CMakeLists.txt:190 (find_package)


 CMake Warning at /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:767
 (message):
   Imported targets not available for Boost version 106501
 Call Stack (most recent call first):
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:871
 (_Boost_COMPONENT_DEPENDENCIES)
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:1501
 (_Boost_MISSING_DEPENDENCIES)
   apps/Launcher/ext/libtorrent/CMakeLists.txt:190 (find_package)



 CMake Warning at /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:767
 (message):
   Imported targets not available for Boost version 106501
 Call Stack (most recent call first):
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:871
 (_Boost_COMPONENT_DEPENDENCIES)
   /usr/local/Cellar/cmake/3.9.2/share/cmake/Modules/FindBoost.cmake:1501
 (_Boost_MISSING_DEPENDENCIES)
   apps/Launcher/ext/libtorrent/CMakeLists.txt:190 (find_package)
```

But when attempting build I get a build failure:

```
/Users/username/git/OpenSpace/apps/Launcher/ext/libtorrent/include/libtorrent/export.hpp:37:12: 'boost/config/select_compiler_config.hpp' file not found
```

Seems to be coming from export.hpp on the line that says:

```
#  include <boost/config/select_compiler_config.hpp>
```

Not 100% Sure where to specify this file.  I also don't see any place to specify boost config or select_compiler_config in cmake.

Thanks for any help in getting this compiled.  Let me know if I can provide any other information.

**Comment by zeninfinity** [comment]
I can't find anywhere to specify the location of boost.  I do see the file is located locally at:  /usr/local/Cellar/boost/1.65.1/include/boost/config/detail/select_compiler_config.hpp but I do not know how to let cmake know where it is...

**Comment by alexanderbock** [comment]
Hi!

According to https://stackoverflow.com/questions/42123509/cmake-finds-boost-but-the-imported-targets-not-available-for-boost-version , that error message might mean that your CMake version is too old.  Can you try installing the newest CMake version, clean your build directory, and try again?

**Comment by zeninfinity** [comment]
Hi Alexander,

Thanks for the response.  

I have installed cmake version 3.9.20170912-g3ea87 from source.    I am using Boost 1.65.1.  I now get 1 error, the same error as above.  It seems to be from the second line from this code:

```
#if !defined(BOOST_COMPILER_CONFIG) && !defined(BOOST_NO_COMPILER_CONFIG)
#  include <boost/config/select_compiler_config.hpp>
#endif
#ifdef BOOST_COMPILER_CONFIG
#  include BOOST_COMPILER_CONFIG
#endif

#if !defined(BOOST_PLATFORM_CONFIG) && !defined(BOOST_NO_PLATFORM_CONFIG)
#  include <boost/config/select_platform_config.hpp>
#endif
#ifdef BOOST_PLATFORM_CONFIG
#  include BOOST_PLATFORM_CONFIG
#endif
```

Any next suggestions?    Can I specify this location/config?  Can I change it to something else?  Anything you can do to help me install this project would be great.  I'm psyched to start using it.

**Comment by zeninfinity** [comment]
I deleted and recreated the build directory, reran ccmake, specificed my gdal locations `/usr/local/Cellar/gdal2/2.2.1_3/bin/gdal-config`, and generated the new config file.  Building still fails.

**Comment by alexanderbock** [comment]
Can you check if the file actually exists for you?  I just checked on my Mac and found it in `/opt/local/include/boost/config/select_platform_config.hpp`.  If the file is there it is likely to be a problem with the include paths (for example `BOOST_ROOT` in CMake being set to a wrong value).  If the file is not there it might be a faulty boost installation?

Judging by the first error that mentioned
`Imported targets not available for Boost version 106501` I would guess it would be the first

**Comment by zeninfinity** [comment]
Thanks again for the response.  I have found the boost file at:

`/usr/local/Cellar/boost/1.65.1/include/boost/config/detail/select_platform_config.hpp`

When running `ccmake -G Xcode ..` from the build directory I do not see a section for BOOST_ROOT under the "B"s:

```...
ASSIMP_PACKAGE_VERSION           0
ASSIM_BUILD_X_IMPORTER           OFF
BUILD_GMOCK                      OFF
BUILD_GTEST                      ON
...```

Is this set somewhere else?  Should I make a symlink to /opt/local/include/boost/config?

**Comment by zeninfinity** [comment]
Ok, I have attempted to go forward.  I have changed `#  include <boost/config/select_compiler_config.hpp>` to `#  include </usr/local/Cellar/boost/1.65.1/include/boost/config/detail/select_platform_config.hpp>` in the file itself and I no longer get that error.  

Now the build continues but I get other errors:

**Comment by zeninfinity** [comment]
<img width="262" alt="screen shot 2017-10-09 at 5 36 31 pm" src="https://user-images.githubusercontent.com/8867994/31362957-8dc94ffa-ad18-11e7-8d1c-29f217b0fe95.png">

<img width="259" alt="screen shot 2017-10-09 at 5 36 43 pm" src="https://user-images.githubusercontent.com/8867994/31362961-941408dc-ad18-11e7-8e55-654a5c22c899.png">

**Comment by zeninfinity** [comment]
At this point I have no idea to move forward.  Thoughts?  What am I missing?

**Comment by alexanderbock** [comment]
This looks like the GDAL library is not included..  The best way is to install it: `brew tap osgeo/osgeo4mac && brew install gdal2` 

alternatively, you can  disable it in CMake with `OPENSPACE_MODULE_GLOBEBROWSING_USE_GDAL` but you'll lose a lot of functionality.

**Comment by zeninfinity** [comment]
Ok, looks like boost was installed and correct.  Looks like the gdal issue was incorrect location to the library and not the lib/ folder:

```
 GDAL_LIBRARY                     /usr/local/Cellar/gdal2/2.2.2/lib/libgdal.dylib
```

3 more issues with the Release branch:

<img width="272" alt="screen shot 2017-10-10 at 5 31 18 pm" src="https://user-images.githubusercontent.com/8867994/31415467-e8165eee-ade0-11e7-9943-5ece1b38bc21.png">

Any thoughts on these?

**Comment by CarlHostetter-NASA** [comment]
I'm having the same link error with Release:

<img width="400" alt="screen shot 2017-10-11 at 11 11 41 am" src="https://user-images.githubusercontent.com/450870/31449483-0a88090c-ae75-11e7-8973-963947f2876b.png">

macOS Sierra (10.12.6), Xcode 9.0

**Comment by CarlHostetter-NASA** [comment]
OK, I’ve had success in building and running OpenSpace on my Mac. I did so by 1) reverting to a default CMake session (by deleting my build folder), redoing the CMake configure (several times to clear all the red) and Xcode project generation process with (only) OPENSPACE_APPLICATION_OPENSPACE enabled (enabling OPENSPACE_APPLICATION_LAUNCHER resulted in a compiler error). NOTE that I did have to adjust the Qt paths as the Mac build instructions on the Wiki suggest.
